### ISSUE TITLE
Reimplement class balancing strategy

### ISSUE THREAD

**Issue by fqixiang** [asreview/asreview#1928]
## Feature Request

**Is your feature request related to a problem? Please describe.**
See the implemented [double balancing algorithm](https://github.com/asreview/asreview/blob/main/asreview/models/balance/double.py). The point of this algorithm was to weigh relevant and irrelevant documents as a function of the number of both types of documents, and four parameters (a, b, alpha, beta) and then resample these documents according to their weights. The result is a resampled dataset (of the same size to the original) that is then used to train machine learning classifiers. 

While this algorithm has been shown to work well for ASReview projects, there are several considerations:

1. This algorithm is built on existing literature on dealing with class imbalances. 
2. It is not systematically evaluated against other competing algorithms.
3. It has too many parameters, of which b and beta have little influence on the rebalancing outcome. This make the algorithm difficult to interpret and hyperparameter tuining more difficult.
4. The parameter space is mostly unbounded.

**Describe the solution you'd like**
There are several ideas, but in all cases, the solution should not underperform the existing algorithm.

1. Simplify the algorithm to have fewere parameters.
2. Replace the algorithm with literature-based algorithms.
3. Replace the algorithm with only weights and use these weights in a classifier's loss function.

**Describe alternatives you've considered**
See above.

**Teachability, Documentation, Adoption, Migration Strategy**
N.A.

**Comment by fqixiang** [comment]
The full parameter space for the existing algorithm can be found [here](https://github.com/asreview/asreview/blob/751a0863c83162467e98c845cf7ac37620d0bdd7/asreview/models/balance/double.py), with help from the creator of the algorithm - Raoul.

```
"bal_a": hp.lognormal("bal_a", 0, 1),
"bal_alpha": hp.uniform("bal_alpha", 0, 2),
"bal_b": hp.uniform("bal_b", 0, 1),
# "bal_beta": hp.uniform("bal_beta", 0, 2),
```

**Comment by fqixiang** [comment]
Ideas gathered from the last stand-up:

- [ ] Make plots showing how the ratio of relevant to irrelevant documents changes as the parameter values change - in order to study how the existing algorithm behaves. Try multiple benchmark datasets.
- [ ] Research existing algorithms in the literature and compare them against the double balancing algorithm in terms of retrieval quality. 
- [ ] Research the use of weights in loss functions and compare it with our existing and other literature-based balancing algorithms, as well as the combination of the two.

**Comment by J535D165** [comment]
> The full parameter space for the existing algorithm can be found [here](https://github.com/asreview/asreview/blob/751a0863c83162467e98c845cf7ac37620d0bdd7/asreview/models/balance/double.py), with help from the creator of the algorithm - Raoul.
> 
> ```
> "bal_a": hp.lognormal("bal_a", 0, 1),
> "bal_alpha": hp.uniform("bal_alpha", 0, 2),
> "bal_b": hp.uniform("bal_b", 0, 1),
> # "bal_beta": hp.uniform("bal_beta", 0, 2),
> ```

Yes, this has been removed from the main branch, as we do hyperparameter tuning in another repo now.
